manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -181,23 +187,47 @@ public BridgeResponse toResponse(Bridge bridge) {
         return response;
     }
 
-    private void createTopicAndServiceAccount(String bridgeId) {
+    private void createTopicAndGrantAccessFor(String bridgeId) {
         if (!rhoasEnabled) {
             return;
         }
-        String topicName = String.format("ob-%s", bridgeId);
-        String serviceAccountName = String.format("ob-%s-consumer", bridgeId);
-        TopicAndServiceAccountRequest request = new TopicAndServiceAccountRequest(topicName, serviceAccountName);
         try {
-            rhoasClient.createTopicAndConsumerServiceAccount(request).await().atMost(Duration.ofSeconds(rhoasTimeout));
+            NewTopicInput newTopicInput = new NewTopicInput()
+                    .name(topicNameFor(bridgeId))
+                    .settings(new TopicSettings().numPartitions(1));
+
+            rhoasClient.createTopicAndGrantAccess(newTopicInput, rhoasOpsAccountClientId, RhoasTopicAccessType.CONSUMER_AND_PRODUCER)
+                    .await().atMost(Duration.ofSeconds(rhoasTimeout));
         } catch (CompletionException e) {
-            String msg = "Failed creating topic and service account for bridge " + bridgeId;
+            String msg = "Failed creating topic and granting access for bridge " + bridgeId;
             LOGGER.warn(msg, e);
             throw new InternalPlatformException(msg, e);
         } catch (TimeoutException e) {
-            String msg = "Timeout reached while creating topic and service account for bridge " + bridgeId;
+            String msg = "Timeout reached while creating topic and granting access for bridge " + bridgeId;
             LOGGER.warn(msg, e);
             throw new InternalPlatformException(msg, e);
         }
     }
+
+    private void deleteTopicAndRevokeAccessFor(String bridgeId) {
+        if (!rhoasEnabled) {
+            return;
+        }
+        try {
+            rhoasClient.deleteTopicAndRevokeAccess(topicNameFor(bridgeId), rhoasOpsAccountClientId, RhoasTopicAccessType.CONSUMER_AND_PRODUCER)
+                    .await().atMost(Duration.ofSeconds(rhoasTimeout));
+        } catch (CompletionException e) {
+            String msg = "Failed deleting topic and revoking access for bridge " + bridgeId;
+            LOGGER.warn(msg, e);
+            throw new InternalPlatformException(msg, e);
+        } catch (TimeoutException e) {
+            String msg = "Timeout deleting topic and revoking access for bridge " + bridgeId;
+            LOGGER.warn(msg, e);
+            throw new InternalPlatformException(msg, e);
+        }
+    }
+
+    private String topicNameFor(String bridgeId) {
+        return String.format("ob-%s", bridgeId);
```

Comment:
@kostola Does this need to be stored on the `Bridge` entity when we persist it to the DB? We need to be able to communicate the details of the Topic chosen to the Shard as part of https://issues.redhat.com/browse/MGDOBR-171

Reply:
I discussed this with @r00ta, who is going to implement MGDOBR-171, and we agreed that it's not needed to store this yet because it can be retrieved from the bridge id.

Reply:
@r00ta @kostola Just to be sure: The Manager will still expose the topic name to the Shard via our `ShardBridgesSyncAPI`, but because the topic name can be inferred from `ob- + $bridge.getId()`, we're saying we don't need to persist it in the DB.

Am I correct?

Reply:
afaik you're correct @robpblake :+1: